src/geometry.h

```diff
@@ -663,214 +663,192 @@ namespace Faunus {
                 }
                 return *this;
             }
+        };
 
-            inline void boundary(Point &a) const override {
-                const auto &boundary_conditions = geometry->boundary_conditions;
-                if (boundary_conditions.coordinates == ORTHOGONAL) {
-                    if (boundary_conditions.direction.x() == PERIODIC) {
-                        if (std::fabs(a.x()) > len_half.x())
-                            a.x() -= len.x() * anint(a.x() * len_inv.x());
-                    }
-                    if (boundary_conditions.direction.y() == PERIODIC) {
-                        if (std::fabs(a.y()) > len_half.y())
-                            a.y() -= len.y() * anint(a.y() * len_inv.y());
-                    }
-                    if (boundary_conditions.direction.z() == PERIODIC) {
-                        if (std::fabs(a.z()) > len_half.y())
-                            a.z() -= len.z() * anint(a.z() * len_inv.z());
-                    }
-                } else {
-                    geometry->boundary(a);
+        inline void Chameleon::_setLength(const Point &l) {
+            len = l;
+            len_half = l * 0.5;
+            len_inv = l.cwiseInverse();
+        }
+
+        inline double Chameleon::getVolume(int dim) const {
```

Comment:
getVolume(), setVolume(), randompos() etc. are not speed limiting so why suggest inlining?

Reply:
As they are one-line wrappers only, hence to remove the possible (unimportant) overhead. Just a matter of habit. Can un-inline and move to cpp.

Reply:
sure, it's a minor detail just to keep the exe size down and leave room for inlined functions of importance. I merge now and you can always revisit it later.